autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingHandler.java

```diff
@@ -45,60 +47,72 @@
+            if (hostGroupNodeCount != desiredNodeCount) {
+                scalingRequest = (Runnable) applicationContext.getBean("ScalingRequest", cluster, policy,
+                        hostGroupNodeCount, desiredNodeCount, List.of());
+            }
         }
-    }
 
-    private long getRemainingCooldownTime(Cluster cluster) {
-        long coolDown = cluster.getCoolDown();
-        long lastScalingActivity = cluster.getLastScalingActivity();
-        return lastScalingActivity == 0L ? 0L : (coolDown * TimeUtil.MIN_IN_MS) - (System.currentTimeMillis() - lastScalingActivity);
+        if (scalingRequest != null) {
+            executorService.submit(scalingRequest);
+            rejectedThreadService.remove(cluster.getId());
+            cluster.setLastScalingActivityCurrent();
+            clusterService.save(cluster);
+        } else {
+            LOGGER.debug("No scaling activity required for cluster crn {}", cluster.getStackCrn());
+        }
     }
 
-    private int getDesiredNodeCount(Cluster cluster, ScalingPolicy policy, int totalNodes) {
+    private int getDesiredNodeCount(Cluster cluster, ScalingPolicy policy, int hostGroupNodeCount) {
         int scalingAdjustment = policy.getScalingAdjustment();
-        int desiredNodeCount;
+        int desiredHostGroupNodeCount;
         switch (policy.getAdjustmentType()) {
             case NODE_COUNT:
-                desiredNodeCount = totalNodes + scalingAdjustment;
+            case LOAD_BASED:
+                desiredHostGroupNodeCount = hostGroupNodeCount + scalingAdjustment;
                 break;
             case PERCENTAGE:
-                desiredNodeCount = totalNodes
-                        + (int) (ceil(totalNodes * ((double) scalingAdjustment / ClusterUtils.MAX_CAPACITY)));
+                desiredHostGroupNodeCount = hostGroupNodeCount
+                        + (int) (ceil(hostGroupNodeCount * ((double) scalingAdjustment / ClusterUtils.MAX_CAPACITY)));
                 break;
             case EXACT:
-                desiredNodeCount = policy.getScalingAdjustment();
+                desiredHostGroupNodeCount = policy.getScalingAdjustment();
                 break;
             default:
-                desiredNodeCount = totalNodes;
+                desiredHostGroupNodeCount = hostGroupNodeCount;
         }
-        int minSize = cluster.getMinSize();
-        int maxSize = cluster.getMaxSize();
-        return desiredNodeCount < minSize ? minSize : desiredNodeCount > maxSize ? maxSize : desiredNodeCount;
+        int minSize = cluster.getHostGroupMinSize();
+        int maxSize = cluster.getHostGroupMaxSize();
+
+        return desiredHostGroupNodeCount < minSize ? minSize : desiredHostGroupNodeCount > maxSize ? maxSize : desiredHostGroupNodeCount;
     }
 
+    private int getHostGroupNodeCount(Cluster cluster, ScalingPolicy policy) {
+        StackV4Response stackV4Response = cloudbreakCommunicator.getByCrn(cluster.getStackCrn());
```

Comment:
Any chance this can be passed in to ScalingHandler? May not be possible for anything other than the YARNLoadEvaluator.

Reply:
have added cacheable to stackv4response, so it is cached for 5 mins.

Reply:
Can the time be adjusted to a lower value? 5 minutes can lead to incorrect information on the number of nodes. Potentially spanning across multiple scale events.

Reply:
agree, removed the cache and used single call for a single evaluation cycle.